src/Middleware/HttpCache.php

```diff
@@ -96,16 +98,12 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
         }
 
         $cacheIsValid = $this->validateCache($request, $lastModified, $etag);
+        $response = $handler->handle($request);
+
         if ($cacheIsValid) {
-            return $this->responseFactory
```

Comment:
@rustamwin would you please verify that the change is correct?

Reply:
That is correct, yes.